packages/aws-cdk-lib/aws-sns/lib/topic.ts

```diff
@@ -154,13 +154,14 @@ export class Topic extends TopicBase {
    * @param scope The parent creating construct
    * @param id The construct's name
    * @param topicArn topic ARN (i.e. arn:aws:sns:us-east-2:444455556666:MyTopic)
+   * @param contentBasedDeduplication If content-based deduplication is enabled
    */
-  public static fromTopicArn(scope: Construct, id: string, topicArn: string): ITopic {
+  public static fromTopicArn(scope: Construct, id: string, topicArn: string, contentBasedDeduplication?: boolean): ITopic {
```

Comment:
🤔 Not sure about this approach.
Do we have a way to detect if `contentBasedDeduplication` is enabled from the `topicArn`?
If not, I'd prefer to create a separate `fromTopicAttributes` method to address the use case as done in other constructs.

Reply:
Yeah let me fix this....